### PR TITLE
testing(): add object.get tests for multihash string

### PR DESCRIPTION
**Pull request by kenshyx**
These tests will pass after https://github.com/ipfs/js-ipfs/pull/849 is merged

### CODE REVIEW

File: src/object.js [hunk 1]
```diff
@@ -781,14 +789,14 @@ module.exports = (common) => {
...
+      it('object.get', () => {
         const testObj = {
           Data: new Buffer('get test object'),
           Links: []
         }
 
-        ipfs.object.put(testObj).then((node1) => {
-          ipfs.object.get(node1.multihash).then((node2) => {
+        return ipfs.object.put(testObj).then((node1) => {
+          return ipfs.object.get(node1.multihash).then((node2) => {
```
Comment (diasdavid) [inline src/object.js]: @kenshyx all of these tests are written to test the callback API. Check at the bottom the section to test de promises API.

We always test in depth through the Callback API.

File: src/object.js [hunk 2]
```diff
@@ -797,7 +805,10 @@ module.exports = (common) => {
 
             expect(node1.data).to.deep.equal(node2.data)
             expect(node1.links).to.deep.equal(node2.links)
-            done()
+            // get object from ipfs multihash string
+            return ipfs.object.get(node1.toJSON().multihash).then((node2) => {
+              expect(node2).to.exist()
+            });
```
Comment (diasdavid) [inline src/object.js]: Make this a separate test.

File: src/object.js (hunk 1 above)
Comment (kenshyx) [inline src/object.js]: I don't understand, those lines are under `promise API` :)

**Comment by diasdavid** [comment]
Oh, I just realized I didn't read correctly where the changes were being added and in fact, you were doing the right thing! It seems that all of the other Promise API tests are finishing through calling done instead of returning a promise, which is bad because errors are not caught.

I'm sorry for the confusion @kenshyx, would you like to fix all the Promise API tests? If not I can do it as my redemption :)

**Comment by kenshyx** [comment]
Yes, I'll do it.

**Comment by kenshyx** [comment]
Let me know if there is something else I have to update @diasdavid

**Pull request merged**